Dexterous-Rambutan/battle-code
# Feature room queue

#### What's this PR do?

Matchmaking system complete:
Any two players who enter multi-player arena, in queuing order, are placed into the same room. Server will emit socket event 'start' to signify that both players have been put into the room
#### What are the important parts of the code?

server/server.js lines 52 - 80
#### How should this be tested by the reviewer?

Start two seperate browsers and both enter into multi-player arena, and should expect an error message on the client browser, coming from app/actions/arenaAction.js line 15

``` javascript
var getProblem = function(payload){
  return function(dispatch){
    $.ajax({
      method: 'GET',
      url: '/api/challenges/:' + payload.challenge_id,
      dataType: 'json',
      cache: false,
      success: function(data){
          dispatch({
            type: actions.GET_PROBLEM_SUCCESS,
            payload: data
          });
      },
      error: function(error){
        dispatch({
          type: actions.GET_PROBLEM_ERROR
        });
      }
    })
  }
};
```

It will say "cannot get challenge_id of undefined"

This happens because the socket emitting 'start' is supposed to also send the challenge along with, however currently there is no such database controller to retrieve a challenge unique to both players
#### Is any other information necessary to understand this?
#### What are the relevant tickets? (Please add `closes`, `refs`, etc)

Refs #49 
#### Screenshots (if appropriate)
